app/src/main/java/com/ichi2/apisample/helper/AnkiDroidHelper.java

```diff
@@ -135,6 +135,98 @@ public Long addNewCustomModel(String modelName, String[] fields, String[] cards,
         return getApi().addNewCustomModel(modelName, fields, cards, qfmt, afmt, css, null, null);
     }
 
+    public boolean checkCustomModel(long modelId, String[] fields, String[] cards, String[] qfmt, String[] afmt, String css) {
+        Uri modelUri = Uri.withAppendedPath(FlashCardsContract.Model.CONTENT_URI, String.valueOf(modelId));
+        Cursor cursor = mResolver.query(modelUri, null, null, null, null);
+        cursor.moveToNext();
```

Comment:
Will this work correctly if there is no such a model at all?

Reply:
Nice catch. I've checked the case with invalid `modelId` - it results in a NullPointerException on AnkIDorid's ContentResolver (line 140). For this and similar resolver calls, I've added rethrowing exception as IllegalArgumentException. Please let me know if you have a better suggestion.